core/src/main/scala/epollcat/internal/ch/SocketHelpers.scala

```diff
@@ -120,6 +136,56 @@ private[ch] object SocketHelpers {
     new InetSocketAddress(inetAddr, port)
   }
 
+  def toInet6SocketAddress(
+      addr: Ptr[posix.netinet.in.sockaddr_in6]
+  ): InetSocketAddress = {
+    val port = posix.arpa.inet.htons(addr.sin6_port).toInt
```

Comment:
Please see htons comment above.  
I think a payoff from passing the considered argument types
can be reaped.